components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/management/v1/service/OrganizationManagementService.java

```diff
@@ -915,6 +943,61 @@ private OrganizationMetadata getOrganizationMetadataResponse(Organization organi
+                                                             List<String> metaAttributes, boolean recursive)
+            throws OrganizationManagementServerException {
+
+        MetaAttributesResponse metaAttributesResponse = new MetaAttributesResponse();
+
+        if (limit != 0 && CollectionUtils.isNotEmpty(metaAttributes)) {
+            boolean hasMoreItems = metaAttributes.size() > limit;
+            boolean needsReverse = StringUtils.isNotBlank(before);
+            boolean isFirstPage = (StringUtils.isBlank(before) && StringUtils.isBlank(after)) ||
+                    (StringUtils.isNotBlank(before) && !hasMoreItems);
+            boolean isLastPage = !hasMoreItems && (StringUtils.isNotBlank(after) || StringUtils.isBlank(before));
+
+            String url = "?limit=" + limit + "&recursive=" + recursive;
+            if (StringUtils.isNotBlank(filter)) {
+                try {
+                    url += "&filter=" + URLEncoder.encode(filter, StandardCharsets.UTF_8.name());
+                } catch (UnsupportedEncodingException e) {
+                    LOG.error("Server encountered an error while building pagination URL for the response.", e);
+                    Error error = OrganizationManagementEndpointUtil.getError(
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getCode(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getMessage(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getDescription());
+                    throw new OrganizationManagementEndpointException(Response.Status.INTERNAL_SERVER_ERROR, error);
+                }
+            }
+            if (hasMoreItems) {
+                metaAttributes.remove(metaAttributes.size() - 1);
+            }
+            if (needsReverse) {
+                Collections.reverse(metaAttributes);
+            }
+            if (!isFirstPage) {
+                String encodedString = Base64.getEncoder().encodeToString(metaAttributes.get(0)
+                        .getBytes(StandardCharsets.UTF_8));
+                Link link = new Link();
+                link.setHref(URI.create(OrganizationManagementEndpointUtil.buildURIForPagination(url) + "&before="
+                        + encodedString));
+                link.setRel("previous");
+                metaAttributesResponse.addLinksItem(link);
+            }
+            if (!isLastPage) {
+                String encodedString = Base64.getEncoder().encodeToString(metaAttributes.get(metaAttributes.size() - 1)
+                        .getBytes(StandardCharsets.UTF_8));
+                Link link = new Link();
+                link.setHref(URI.create(OrganizationManagementEndpointUtil.buildURIForPagination(url) + "&after="
+                        + encodedString));
+                link.setRel("next");
+                metaAttributesResponse.addLinksItem(link);
```

Comment:
As discussed offline the cursor value needs to be changed to build based on org creation time.
The value used for the cursor should be unique and sequential

Reply:
We have decided offline to proceed with using the meta attribute key for pagination cursor values. The meta attribute key, retrieved from the database, consists of a distinct set of strings ordered in dictionary order, which can be used as the cursor.